src/core/coap/coap.cpp

```diff
@@ -962,6 +962,33 @@ uint32_t ResponsesQueue::ResponseMetadata::GetRemainingTime(void) const
     return (mDequeueTime > now) ? mDequeueTime - now : 0;
 }
 
+/// Return product of @p aValueA and @p aValueB if no overflow otherwise 0.
+static uint32_t Multiple(uint32_t aValueA, uint32_t aValueB)
+{
+    uint32_t result = aValueA * aValueB;
+
+    return (result / aValueA == aValueB) ? result : 0;
+}
+
+bool TxParameters::Validate(const otCoapTxParameters &aTxParameters)
+{
+    bool rval = false;
+
+    if (aTxParameters.mAckRandomFactorNumerator >= aTxParameters.mAckRandomFactorDenominator &&
+        aTxParameters.mAckTimeout >= OT_COAP_MIN_ACK_TIMEOUT && aTxParameters.mMaxRetransmit <= OT_COAP_MAX_RETRANSMIT)
+    {
+        // Calulate exchange lifetime step by step and verify no overflow.
+        uint32_t tmp = Multiple(aTxParameters.mAckTimeout, (1U << (aTxParameters.mMaxRetransmit + 1)) - 1);
+
+        tmp /= aTxParameters.mAckRandomFactorDenominator;
+        tmp = Multiple(tmp, aTxParameters.mAckRandomFactorNumerator);
```

Comment:
This is ok but should be aware that in some cases the order (whether multiply first or divide first) may matter (result in different rounding happening). I don't think it really impacts though.
(1001 * 8 / 3)  = 2669
(1001 / 3 * 8)  = 2664

Reply:
Yes, this is intentionally as suggested by @dismirlian